packages/core/src/common/keybinding.ts

```diff
@@ -132,50 +144,83 @@ export class KeybindingRegistry {
-        commands.push(binding);
-        this.commands[commandId] = bindings;
+        /* Keep a mapping of the Command -> Key mapping.  */
+        if (!this.isPseudoCommand(binding.commandId)) {
+            const commands = this.commands[binding.commandId] || [];
+            commands.push(binding);
+            this.commands[binding.commandId] = bindings;
+        }
     }
 
     /**
-     * The `active` flag with `false` could come handy when we do not want to check whether the command is currently active or not.
-     * For instance, when building the main menu, it could easily happen that the command is not yet active (no active editors and so on)
-     * but still, we have to build the key accelerator.
+     * Get the keybindings associated to commandId.
      *
-     * @param commandId the unique ID of the command for we the associated ke binding are looking for.
-     * @param options if `active` is false` then the availability of the command will not be checked. Default is `true`
+     * @param commandId The ID of the command for which we are looking for keybindings.
+     * @param options An instance of #GetKeyBindingsOptions.`
      */
-    getKeybindingForCommand(commandId: string, options: { active: boolean } = ({ active: true })): Keybinding | undefined {
-        const bindings = this.commands[commandId];
-        if (!bindings) {
-            return undefined;
-        }
-        if (!options.active) {
-            return bindings[0];
+    getKeybindingsForCommand(commandId: string, options: GetKeyBindingsOptions = GetKeyBindingsOptionsDefault): Keybinding[] {
+        let bindings = this.commands[commandId] || [];
+
+        if (options.activeOnly) {
+            bindings = bindings.filter((k: Keybinding) => this.isActive(k));
         }
-        return bindings.find(this.isActive.bind(this));
+
+        return bindings;
     }
 
     /**
-     * @param keyCode the key code of the binding we are searching.
+     * Get the list of keybindings matching keyCode.  The list is sorted by
+     * priority (see #sortKeybindingsByPriority).
+     *
+     * @param keyCode The key code for which we are looking for key bindings.
+     * @param options An instance of #GetKeyBindingsOptions.
      */
-    getKeybindingForKeyCode(keyCode: KeyCode, options: { active: boolean } = ({ active: true })): Keybinding | undefined {
-        const bindings = this.keybindings[keyCode.keystroke];
-        if (!bindings) {
-            return undefined;
-        }
-        if (!options.active) {
-            return bindings[0];
+    getKeybindingsForKeyCode(keyCode: KeyCode, options: GetKeyBindingsOptions = GetKeyBindingsOptionsDefault): Keybinding[] {
+        let bindings = this.keybindings[keyCode.keystroke] || [];
+
+        if (options.activeOnly) {
+            bindings = bindings.filter((k: Keybinding) => this.isActive(k));
         }
-        return bindings.find(this.isActive.bind(this));
+
+        this.sortKeybindingsByPriority(bindings);
```

Comment:
I would consider an overwritten keybinding to be `inactive` so they should be filtered.

Reply:
Ok, but we should first add documentation to the isActive method to know what it really means for a keybinding to be "active".  Right now, it maps 1:1 to whether the associated command is active.  I don't mind if it changes, but it should be clarified.

Reply:
Sure, please add documention.